receiver/varnishreceiver/config.go

```diff
@@ -46,3 +53,16 @@ func (c *Config) Validate() error {
 
 	return nil
 }
+
+// SetDefaultHostname sets a default hostname if config InstanceName is empty.
+func (c *Config) SetDefaultHostname() error {
+	if c.InstanceName == "" {
+		hostname, err := os.Hostname()
```

Comment:
Does this happen before or after config validation and get passed into the `-n` argument? If it does is this always valid?

Reply:
this was in the wrong place. I need to preserve the was InstanceName, now CacheDir and so I moved this to the scraper since it is relevant to the scraper.